service/routes.py

```diff
@@ -59,159 +59,200 @@ def index():
     """
-    Creates an inventory item
+    InventoryResource class
 
-    This endpoint will create an item based on the data in the body that is posted
+    Allows the manipulation of a single Inventory Item
+    GET /inventory/{inventory_id} - Returns an Item with the inventory_id
+    PUT /inventory/{inventory_id} - Update an Item with the inventory_id
+    DELETE /inventory/{inventory_id} -  Deletes an Item with the inventory_id
     """
-    app.logger.info("Request to create an inventory item")
-    check_content_type("application/json")
-    item = Inventory()
-    item.deserialize(request.get_json())
-    item.create()
-    message = item.serialize()
-    location_url = url_for("get_inventory", inventory_id=item.id, _external=True)
 
-    app.logger.info("Inventory item named [%s] with ID [%s] created.", item.name, item.id)
-    return jsonify(message), status.HTTP_201_CREATED, {"Location": location_url}
+    ######################################################################
+    # RETRIEVE AN INVENTORY ITEM
+    ######################################################################
+    @api.doc('get_inventory_items')
+    @api.response(404, 'Inventory Item not found')
+    @api.marshal_with(inventory_model)
+    def get(self, inventory_id):
+        """
+        Retrieve a single Inventory
+
+        This endpoint will return a Inventory based on it's id
+        """
+        app.logger.info("Request for item with id: %s", inventory_id)
+        inventory = Inventory.find(inventory_id)
+        if not inventory:
+            abort(status.HTTP_404_NOT_FOUND, f"Inventory with id '{inventory_id}' was not found.")
+
+        app.logger.info("Returning item: %s", inventory.name)
+        return inventory.serialize(), status.HTTP_200_OK
+
+    ######################################################################
+    #  UPDATE AN INVENTORY ITEM
+    ######################################################################
+
+    @api.doc('update_inventory_items')
+    @api.response(404, 'Pet not found')
+    @api.response(400, 'The posted Item data was not valid')
+    @api.expect(inventory_model)
+    @api.marshal_with(inventory_model)
+    def put(self, inventory_id):
+        """
+        Updating an inventory item
+
+        This endpoint will update an item based on the data in the body that is posted
+        """
+
+        app.logger.info("Request to update an inventory item with inventory_id:%s", inventory_id)
+        check_content_type("application/json")
+        item = Inventory.find(inventory_id)
+        if not item:
+            abort(status.HTTP_404_NOT_FOUND, f"Item with inventory_id: {inventory_id} not found")
+        try:
+            item.deserialize(request.get_json())
+            item.id = inventory_id
+            item.update()
+        except DataValidationError:
+            abort(status.HTTP_400_BAD_REQUEST, "Malformed request")
+
+        return item.serialize(), status.HTTP_200_OK
```

Comment:
You should access the data using `api.payload`
You can also get rid of the try except block since you have provided a 400 response in the decorator